src/scroll-timeline-base.js

```diff
@@ -296,25 +296,82 @@ export class ScrollTimeline {
   }
 }
 
+// Methods for calculation of the containing block.
+// See https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block.
+
+function findClosestAncestor(element, matcher) {
+  let candidate = element.parentElement;
+  while(candidate != null) {
+    if (matcher(candidate))
+      return candidate;
+    candidate = candidate.parentElement;
+  }
+}
+
+function isBlockContainer(element) {
+  const style = getComputedStyle(element);
+  switch (style.display) {
+    case 'block':
+    case 'inline-block':
+    case 'list-item':
+    case 'table':
+    case 'table-caption':
+    case 'flow-root':
+    case 'flex':
+    case 'grid':
+      return true;
+  }
+
+  return false;
+}
+
+function isAbsoluteOrFixedElementContainer(element) {
+  const style = style.getComputedStyle(element);
+  if (style.transform != 'none' || style.perspective != 'none')
+    return true;
+
+  if (style.willChange == 'transform' || style.willChange == 'perspective')
+    return true;
+
+  if (style.filter != 'none' || style.willChange == 'filter')
+    return true;
+
+  if (style.backdropFilter != 'none')
+    return true;
+
+  return false;
+}
+
+function getContainingBlock(element) {
+  switch (getComputedStyle(element).position) {
+    case 'static':
+    case 'relative':
+    case 'sticky':
+      return findClosestAncestor(element, isBlockContainer);
+
+    case 'absolute':
+    case 'fixed':
+      return findClosestAncestor(element, isAbsoluteOrFixedElementContainer);
```

Comment:
This skips step 2:
> If the position property is absolute, the containing block is formed by the edge of the padding box of the nearest ancestor element that has a position value other than static (fixed, absolute, relative, or sticky).

tldr; if position absolute, you need to consider the above condition or any fixed container condition.